pkg/cfn/components/composer.go

```diff
@@ -1139,3 +1139,92 @@ func (c *BlockstoragePolicyComposer) ManagedPolicy() *managedpolicy.ManagedPolic
 
 	return managedpolicy.New("BlockstoragePolicy", policyName, policyDesc, d)
 }
+
+// CloudwatchDatasourcePolicyComposer contains state for building
+// a managed iam policy that allows grafana to read cloudwatch metrics
+// and logs
+// - https://grafana.com/docs/grafana/latest/datasources/cloudwatch/#iam-policies
+type CloudwatchDatasourcePolicyComposer struct {
+	Repository  string
+	Environment string
+}
+
+// NewCloudwatchDatasourcePolicyComposer returns an initialised cloudwatch datasource policy composer
+func NewCloudwatchDatasourcePolicyComposer(repository, env string) *CloudwatchDatasourcePolicyComposer {
+	return &CloudwatchDatasourcePolicyComposer{
+		Repository:  repository,
+		Environment: env,
+	}
+}
+
+// Compose builds the policy and returns the result
+func (c *CloudwatchDatasourcePolicyComposer) Compose() (*cfn.Composition, error) {
+	p := c.ManagedPolicy()
+
+	return &cfn.Composition{
+		Outputs:   []cfn.StackOutputer{p},
+		Resources: []cfn.ResourceNamer{p},
+	}, nil
+}
+
+// ManagedPolicy returns the policy
+func (c *CloudwatchDatasourcePolicyComposer) ManagedPolicy() *managedpolicy.ManagedPolicy {
+	policyName := fmt.Sprintf("okctl-%s-%s-CloudwatchDatasourceServiceAccountPolicy", c.Repository, c.Environment)
+	policyDesc := "Service account policy for reading cloudwatch metrics and logs from grafana"
+
+	d := &policydocument.PolicyDocument{
+		Version: policydocument.Version,
+		Statement: []policydocument.StatementEntry{
+			{
+				Effect: policydocument.EffectTypeAllow,
+				Action: []string{
+					"cloudwatch:DescribeAlarmsForMetric",
```

Comment:
To me, these actions are Magic Strings :tm:, so I always greatly appreciate if it's documented where these values come from.

In this case, I guess we can comment that they can be found here: https://grafana.com/docs/grafana/latest/datasources/cloudwatch/

Reply:
Totally agreed, and I actually have it: https://github.com/oslokommune/okctl/pull/375/files/52f84c97d1dd1d3bee73c894fded138b470ab8b9#diff-2927cfffd116f3acaf9c61959ce59f8a23543a812fd8f63edf8551d7c7a6d31eR1146

Maybe it should have been closer to the strings themselves?

Reply:
Aah right, I scrolled too quickly past it. Maybe :shrug: , but it makes sense to put on the struct doc as well.